### PR TITLE
feat: add data.load_balancer_type(s) data sources

### LINKED ISSUE #945: [Feature]: add data source `load_balancer_types`
### What whould you like to see?

I would like to see a data source to read all existing lb types from the API (like it is already implemented for locations, datacenters and servers).

I don't think there is much more that can be said about this and no further examples can be given. The data sources mentioned above serve as an example.

### PR DESCRIPTION
**Pull request by jooola**
Closes https://github.com/hetznercloud/terraform-provider-hcloud/issues/945

### CODE REVIEW

**Comment by jooola** [comment]
This data source is almost finished, it is missing the following:
- the hcloud-go library does not provide the `deprecated` string field for the load balancer type https://docs.hetzner.cloud/#load-balancer-types-get-a-load-balancer-type, and I am not sure if we want to wait for the backend to implement the new deprecation object before we at it here.
- the documentation is written, but I am not sure if I missed anything.
- I am not sure what the `-x` suffix is used for in the docs sidebar `sidebar_current` template?

**Comment by jooola** [comment]
> the hcloud-go library does not provide the deprecated string field for the load balancer type https://docs.hetzner.cloud/#load-balancer-types-get-a-load-balancer-type, and I am not sure if we want to wait for the backend to implement the new deprecation object before we at it here.

I think we should not implement an attribute that will probably be deprecated soon. I'd therefore merge this without the deprecation information.

> the documentation is written, but I am not sure if I missed anything.

Everything should be written.

> I am not sure what the -x suffix is used for in the docs sidebar sidebar_current template?

Unrelated to this PR.


This PR is ready for review.

File: internal/loadbalancertype/data_source.go [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+		"id": schema.Int64Attribute{
+			Optional: true,
+			Computed: true,
+		},
+		"name": schema.StringAttribute{
+			Optional: true,
+			Computed: true,
+		},
+		"description": schema.StringAttribute{
+			Computed: true,
```
Comment (apricote) [inline internal/loadbalancertype/data_source.go:75]: Why is `description` the only non-optional attribute?

**Review by apricote — APPROVED** [review]

File: internal/loadbalancertype/data_source.go (hunk 1 above)
Comment (jooola) [reply internal/loadbalancertype/data_source.go:75]: All the fields except id and name should be read only, I updated this.

**Review by apricote — APPROVED** [review]

**Pull request merged**